dvc/repo/checkout.py

```diff
@@ -34,9 +44,19 @@ def _checkout(
 ):
     from dvc.stage import StageFileDoesNotExistError, StageFileBadNameError
 
+    unused = []
+    stats = {
+        "added": [],
+        "deleted": [],
+        "modified": [],
+        "failed": [],
+    }
     if not targets:
         targets = [None]
-        _cleanup_unused_links(self)
+        unused = _get_unused_links(self)
+
+    stats["deleted"] = [_fspath_dir(u, self.root_dir) for u in unused]
+    self.state.remove_links(unused)
```

Comment:
Why is this not under if? Looks fragile this way, we shouldn't call `remove_links()` at all if a particular target is specified.

Reply:
It's empty if there's no argument.

Reply:
Also, I need to check if something is a dir before deleting.

Reply:
There is also an argument, it might be an empty sequence, but it's there.

Reply:
Checkout in dvc:
- checks out particular thing(s) if that things (target) is specified
- if no target is specified it checks out everything and removes anything previously checked out and having no references now.

I.e. if there is target we don't try to change the rest of the workspace, so `.remove_links()` should not be called. The logic of your code is different.

Reply:
If targets are specified, `unused` will be empty and hence, nothing will get removed. If your concern is regarding calling `.remove_links()`, I don't think there's any reason to be defensive.